### PR TITLE
Add new objectFieldOffset for Java 10 and up

### PR DESCRIPTION
**Pull request by theresa-m**
Fixes: #1266

Signed-off-by: Theresa Mammarella <Theresa.T.Mammarella@ibm.com>

### CODE REVIEW

File: jcl/src/java.base/share/classes/jdk/internal/misc/Unsafe.java [hunk 1]
```diff
@@ -851,6 +851,19 @@ private native void copySwapMemory0(Object srcObj, long srcOffset, Object destOb
...
+	/*
+	 * Returns byte offset to field.
+	 * 
+	 * @param class with desired field
+	 * @param string name of desired field
+	 * @return offset to start of class or interface
+	 * 
+	 * @throws IllegalArgumentException if field is static
+	 */
+	private native long objectFieldOffset1(Class c, String fieldName);
```
Comment (DanHeidinga) [inline jcl/src/java.base/share/classes/jdk/internal/misc/Unsafe.java]: `Class` -> `Class<?>`

File: jcl/src/java.base/share/classes/jdk/internal/misc/Unsafe.java [hunk 2]
```diff
@@ -1354,6 +1367,24 @@ public long objectFieldOffset(Field field) {
...
+	 * Returns byte offset to field.
+	 * 
+	 * @param class with desired field
+	 * @param string name of desired field
+	 * @return offset to start of class or interface
+	 * 
+	 * @throws NullPointerException if field parameter is null
+	 * @throws IllegalArgumentException if field is static
+	 */
+	public long objectFieldOffset(Class c, String fieldName) {
```
Comment (DanHeidinga) [inline jcl/src/java.base/share/classes/jdk/internal/misc/Unsafe.java]: `Class` -> `Class<?>`: Avoid raw types.

File: jcl/src/java.base/share/classes/jdk/internal/misc/Unsafe.java (hunk 1 above)
Comment (DanHeidinga) [inline jcl/src/java.base/share/classes/jdk/internal/misc/Unsafe.java]: Where do I find the implementation of `objectFieldOffset1`?  Is that in a separate PR or is this intended to be a stub?

Comment (theresa-m) [reply jcl/src/java.base/share/classes/jdk/internal/misc/Unsafe.java]: It was written in by @JasonFengJ9 in sun_misc_Unsafe.cpp for initial java 10 support. Thanks Jason :)

**Comment by DanHeidinga** [comment]
Jenkins test sanity plinux

**Comment by andrew-m-leonard** [comment]
@DanHeidinga Is this ready to merge?

**Comment by DanHeidinga** [comment]
@andrew-m-leonard Merged.  Thanks for the poke on this.

**Pull request merged**